napi_js_data.md

```diff
@@ -0,0 +1,873 @@
+```C
+napi_status napi_create_number(napi_env env, double value, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: Double-precision value to be represented in JavaScript
+- `[out] result`: A `napi_value` representing a JavaScript Number
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API is used to convert from the C++ double type to the JavaScript 
+Number type.
+
+The JavaScript Number type is described in
+[Section 6.1.6](https://tc39.github.io/ecma262/#sec-ecmascript-language-types-number-type) 
+of the ECMAScript Language Specification.
+
+### *napi_create_string_utf16*
+
+#### Signature
+```C
+napi_status napi_create_string_utf16(napi_env env,
+                                     const char16_t* str,
+                                     size_t length,
+                                     napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  str`: Character buffer representing a UTF16-LE encoded string
+- `[in]  length`: The length of the string in characters, or -1 if it is
+null-terminated.
+- `[out] result`: A `napi_value` representing a JavaScript String
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API creates a JavaScript String object from a UTF16-LE encoded C++ string
+
+JavaScript String type is described in
+[Section 6.1.4](https://tc39.github.io/ecma262/#sec-ecmascript-language-types-string-type) 
+of the ECMAScript Language Specification.
+
+### *napi_create_string_utf8*
+
+#### Signature
+```C
+napi_status napi_create_string_utf8(napi_env env,
+                                    const char* str,
+                                    size_t length,
+                                    napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  s`: Character buffer representing a UTF8-encoded string
+- `[in]  length`: The length of the string in characters, or -1 if it is
+null-terminated.
+- `[out] result`: A `napi_value` representing a JavaScript String
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API creates a JavaScript String object from a UTF8-encoded C++ string
+
+JavaScript String type is described in
+[Section 6.1.4](https://tc39.github.io/ecma262/#sec-ecmascript-language-types-string-type)
+of the ECMAScript Language Specification.
+
+## Functions to convert from N-API to C++ types
+
+### *napi_get_array_length*
+
+#### Signature
+```C
+napi_status napi_get_array_length(napi_env env,
+                                  napi_value value,
+                                  uint32_t* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: napi_value representing the JavaScript Array whose length is
+being queried
+- `[out] result`: uint32 representing length of the array
```

Comment:
Perhaps "&#96;uint32_t&#96; representing...", or "Length of the array as a &#96;uint32_t&#96;"